lib/Dialect/Seq/SeqOps.cpp

```diff
@@ -455,6 +474,49 @@ LogicalResult FirRegOp::canonicalize(FirRegOp op, PatternRewriter &rewriter) {
     return success();
   }
 
+  // For reset-less 1d array registers, replace an uninitialized element with
+  // constant zero. For example, let `r` be a 2xi1 register and its next value
+  // be `{foo, r[0]}`. `r[0]` is connected to itself so will never be
+  // initialized. If we don't enable aggregate preservation, `r_0` is replaced
+  // with `0`. Hence this canonicalization replaces 0th element of the next
+  // value with zero to match the behaviour.
+  if (!op.getReset()) {
+    if (auto arrayCreate = op.getNext().getDefiningOp<hw::ArrayCreateOp>()) {
+      // For now only support 1d arrays.
+      // TODO: Support nested arrays and bundles.
+      if (arrayCreate.getOperands().front().getType().isa<IntegerType>()) {
```

Comment:
I think I have asked this before, but can ArrayCreate take 0 arguments and return a 0-length array?  It might be better to check `op.result().getType().getElementType().isa<IntegerType>()`.

Reply:
It looks currently we can construct 0 arguments ArrayCreate in-memory but it is not allowed to print MLIR format. Either way I agree that `op.result().getType().getElementType().isa<IntegerType>()` would be better, thanks!